### ISSUE TITLE
Valid Datalevin and Datahike queries are reported as :datalog-syntax errors

### ISSUE THREAD

**Issue by whilo** [clj-kondo/clj-kondo#2945]
clj-kondo depends on `datalog-parser` 0.2.30, which predates the `:order-by`, `:having` and `:timeout` clauses that Datalevin and Datahike support. `datalevin.core` and `datahike.api` are both in the list of libraries the linter recognises, so linting ordinary code for either reports errors on queries that run fine.

```clojure
(ns user (:require [datalevin.core :refer [q]]))

(q '[:find ?e ?score :where [?e :item/score ?score] :order-by [?score :desc] :limit 5] db)
;; Cannot parse clause, expected (data-pattern | pred-expr | fn-expr | rule-expr | ...)

(q '{:find [?e] :where [[?e :age 30]] :limit 5} db)
;; Don't know how to create ISeq from: java.lang.Long
```

The first happens because the parser folds a key it does not know into the vector of the preceding key, so an unrecognised `:order-by` surfaces as an error about `:where`. The second because `:limit`/`:offset` assumed the vector syntax and called `first` on the value.

There is a third, specific to Datahike. It pre-installs its bitemporal rules on every query context and rewrites a query calling one to pass them along, so the query as written never binds `%`:

```clojure
(ns user (:require [datahike.api :refer [q]]))

(q '[:find ?s :in $ ?at :where (valid-at ?tx ?at) [?e :emp/salary ?s ?tx true]] db #inst "2024-04-15")
;; Missing rules var '%' in :in
```

That one is reported three times in Datahike's own `valid_time_test.cljc`, on queries whose tests assert they return results.

All three are fixed in datalog-parser 0.2.42. The first two by the parser itself; the third by a new `:implicit-rules?` option a caller passes when it knows the engine pre-installs rules, since a parser cannot know an engine's rule names. Note the artifact also moved group id, `io.replikativ` -> `org.replikativ`.

I maintain datalog-parser, and I have a branch ready with the bump, the option wired up per library (with a `:datalog-syntax {:implicit-rules true|false}` override), and regression tests. Linting datahike, datalevin and datascript with it — 482 files — removes 8 findings and adds none. Happy to send it as a PR if the approach looks right.
